core/runtime/src/main/java/io/quarkus/runtime/LocalesBuildTimeConfig.java

```diff
@@ -44,8 +45,12 @@ public class LocalesBuildTimeConfig {
      * For instance, the Hibernate Validator extension makes use of it.
      * <p>
      * Native-image build uses this property to derive {@code user.language} and {@code user.country} for the application's
-     * runtime.
+     * runtime. Starting with GraalVM for JDK 24 {@code user.language} and {@code user.country} can also be overridden at
+     * runtime, provided the selected locale was included at image build time.
+     * <p>
+     * Defaults to the JVM's default locale if not set. Starting with GraalVM for JDK 24, it defaults to {@code en-US}
+     * for native executables.
      */
     @ConfigItem(defaultValue = DEFAULT_LANGUAGE + "-" + DEFAULT_COUNTRY, defaultValueDocumentation = "Build system locale")
-    public Locale defaultLocale;
+    public Optional<Locale> defaultLocale;
```

Comment:
You make it optional but you still have a `defaultValue` set so I fail to see when it's going to be empty?

You're mentioning an Hibernate Validator issue but it looks like you fixed it? Maybe you forgot to drop it afterwards?

Reply:
Fixed with https://github.com/quarkusio/quarkus/pull/43448/commits/0ea1c6b63a00a491575a4d0da2c01d8cadc6d5cc PTAL (I will squash once you are OK with the changes and CI passes)

Reply:
Ideally, you can remove the sentence starting with `Defaults to...` in the main javadoc as the `defaultValueDocumentation` will be printed in the doc.

Anyway, looks good to me. I let you squash things as you see fit and then we can merge!